include/v8-instance.h

```diff
@@ -29,53 +36,24 @@ class v8_instance {
         });
     }
 
-    bool run_instance() {
-        is_finish.store(false);
-        v8::Locker locker(isolate);
-        v8::Isolate::Scope isolate_scope(isolate);
-        v8::HandleScope handle_scope(isolate);
-        v8::TryCatch try_catch(isolate);
-        v8::Local<v8::Context> local_ctx = v8::Local<v8::Context>::New(isolate, context);
-        v8::Context::Scope context_scope(local_ctx);
-
-        const int argc = 1;
-        auto array = v8::ArrayBuffer::New(isolate, store);
-        v8::Local<v8::Value> argv[argc] = { array };
-        v8::Local<v8::Value> result;
-
-        v8::Local<v8::Function> local_function = v8::Local<v8::Function>::New(isolate, function);
-        if (!local_function->Call(local_ctx, local_ctx->Global(), argc, argv).ToLocal(&result)) {
-            v8::String::Utf8Value error(isolate, try_catch.Exception());
-            std::cout << "Can not run script: " << std::string(*error, error.length()) << std::endl;
-            is_finish.store(true);
-            return false;
-        }
-
-        is_finish.store(true);
-        return true;
-    }
-
-    void wrap_external_memory(char* data_ptr, size_t size) {
-        store = v8::ArrayBuffer::NewBackingStore(data_ptr, size, v8::BackingStore::EmptyDeleter, nullptr);
+    seastar::future<bool> run_instance(v::ThreadPool& thread_pool, int timeout, std::span<char> data) {
+        return seastar::with_semaphore(mtx, 1, [this, &thread_pool, timeout, data](){
+            is_cancel = false;
+            watchdog.rearm(seastar::lowres_clock::time_point(seastar::lowres_clock::now() + std::chrono::seconds(timeout)));
+            return thread_pool.submit([this, data](){
+                run_instance_internal(data);
+            })
+            .then([this] {
```

Comment:
could these two lambdas be merged? we can call run_instance_internal() and then cancel the timer if needed without creating new continuation

Reply:
For that we need use mutex for work with `is_canceled` and `watchdog`. I